protected/migrations/m170823_064156_cera_ethnic_groups.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+class m170823_064156_cera_ethnic_groups extends OEMigration
+{
+	public function safeUp()
+	{
+        /**
+         * @var $ethnicity EthnicGroup
+         * @var $patient Patient
+         */
+	    // Save all ethnic groups so the original values can be versioned.
+        $ethnicGroups = EthnicGroup::model()->findAll();
+        foreach ($ethnicGroups as $ethnicity) {
+            $ethnicity->save();
+        }
+
+	    // Mark all patients whose ethnic group is being removed as having Other ethnicity.
+	    $patients = Patient::model()->findAll();
+	    foreach ($patients as $patient) {
+	        $patient->ethnic_group_id = 16;
+	        if (!$patient->save()) {
+	            throw new CDbException('Unable to save patient data.');
+            }
+        }
+
+        // Remove the unused ethnic groups.
+        $removedEthnicGroups = EthnicGroup::model()->findAllBySql('SELECT * FROM ethnic_group WHERE id BETWEEN 8 and 15');
+	    foreach ($removedEthnicGroups as $ethnicity) {
+            $ethnicity->delete();
+        }
+
+        // Change the text of the remaining ethnic groups.
+        $ethnicGroups = EthnicGroup::model()->findAll();
+	    foreach ($ethnicGroups as $ethnicity) {
+	        switch ($ethnicity->id) {
+                case 1:
+                    $ethnicity->name = 'Australia';
+                    break;
+                case 2:
+                    $ethnicity->name = 'China';
+                    break;
+                case 3:
+                    $ethnicity->name = 'Greece';
+                    break;
+                case 4:
+                    $ethnicity->name = 'Italy';
+                    break;
+                case 5:
+                    $ethnicity->name = 'New Zealand';
+                    break;
+                case 6:
+                    $ethnicity->name = 'United Kingdom';
+                    break;
+                case 7:
+                    $ethnicity->name = 'India';
+                    break;
+                case 16:
+                    $ethnicity->name = 'Other';
+                    break;
+                case 17:
+                    $ethnicity->name = 'Don\'t know';
+                    break;
+                default:
+                    throw new UnexpectedValueException('Invalid ethnic group value.');
+            }
+            $ethnicity->save();
+        }
+	}
+
+	// This migration cannot be migrated down without losing data integrity.
```

Comment:
What does this do when you migrate down this module? Does it throw a "can't be migrated down" error? I've seen other migrations that explicitly throw an exception to prevent migrating down